com.unity.perception/Editor/Character/ModelTestUI.cs

```diff
@@ -0,0 +1,160 @@
+                    GUILayout.Label("Character Tools", EditorStyles.whiteLargeLabel);
+
+                    drawFaceRays = GUILayout.Toggle(drawFaceRays, "Draw Face Rays");
+
+                    var failedBones = new Dictionary<HumanBone, bool>();
+                    var failedPose = new List<GameObject>();
+                                       
+
+
+                    if (GUILayout.Button("Create Nose and Ears", GUILayout.Width(160)))
+                    {
+                        testResults = TestResults.Running;
+
+                        var test = contentTests.CharacterCreateNose(selection, animator, skinnedMeshRenderer, drawFaceRays);
+                        
+
+                        if (test)
+                            testResults = TestResults.Pass;
+                        if (!test)
+                            testResults = TestResults.Fail;
+                    }
+
+                    break;
+
+                case 1:
+
+                    GUILayout.Label("Character Validation", EditorStyles.whiteLargeLabel);
+                    GUILayout.Label(string.Format("Validation for Character : {0}", testResults), EditorStyles.whiteLabel);
+
+                    if (animator != null)
+                    {
+                        GUILayout.Label(string.Format("Character is Human: {0}", animator.avatar.isHuman), EditorStyles.boldLabel);
+                        GUILayout.Label(string.Format("Character is Valid: {0}", animator.avatar.isValid), EditorStyles.boldLabel);
+                    }
+
+                    if (GUILayout.Button("Validate Bones", GUILayout.Width(160)))
+                    {
+                        testResults = TestResults.Running;
+
+                        var test = contentTests.CharacterRequiredBones(animator, out failedBones);
+
+                        if (failedBones.Count > 0)
+                        {
+                            for (int i = 0; i < RequiredBones.Length; i++)
+                            {
+                                for (int b = 0; b < failedBones.Count; b++)
+                                {
+                                    var bone = failedBones.ElementAt(i);
+                                    var boneKey = bone.Key;
+                                    var boneValue = bone.Value;
+
+                                    if (RequiredBones[i] == boneKey.humanName)
+                                    {
+                                        GUILayout.Label(string.Format("Bone {0}: {1}", RequiredBones[i], "Missing"), EditorStyles.boldLabel);
+                                    }
+                                }
+                            }
+                        }
+                        else if (failedBones.Count == 0)
+                        {
+                            GUILayout.Label(string.Format("Required Bones Present : {0}", TestResults.Pass), EditorStyles.whiteLabel);
+                        }
+
+                        if (test)
```

Comment:
this could be reduced to: testResults = test ? TestResults.Pass : TestResults.Fail;